vsphere/internal/helper/folder/folder_helper.go

```diff
@@ -191,6 +191,8 @@ func folderFromObject(client *govmomi.Client, obj interface{}, folderType RootPa
 		p, err = RootPathParticleNetwork.PathFromNewRoot(o.InventoryPath, folderType, relative)
 	case *object.Datastore:
 		p, err = RootPathParticleDatastore.PathFromNewRoot(o.InventoryPath, folderType, relative)
+	case *object.StoragePod:
+		p, err = RootPathParticleDatastore.PathFromNewRoot(o.InventoryPath, folderType, relative)
```

Comment:
It would be nice if they just used some interfaces occasionally :cry:

Reply:
Not 100% ruling out there could be some refactoring here, and `InventoryPath` is part of a `Common` object that everything embeds, but we still need to know what _kind_ of object is being passed in, and there is no common interface for say datastore clusters. In fact, this would not 100% work here even if there was, as for example, `StoragePod` actually would inherit from `Folder` versus `Datastore` - and indeed emebeds that object instead of `Datastore`. :stuck_out_tongue: